block/p2p.go

```diff
@@ -0,0 +1,61 @@
+package block
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dymensionxyz/dymint/p2p"
+	"github.com/dymensionxyz/dymint/types"
+	"github.com/tendermint/tendermint/libs/pubsub"
+)
+
+// onReceivedBlock receives a block received event from P2P, saves the block to a cache and tries to apply the blocks from the cache.
+func (m *Manager) onReceivedBlock(event pubsub.Message) {
+	eventData, ok := event.Data().(p2p.P2PBlockEvent)
+	if !ok {
+		m.logger.Error("onReceivedBlock", "err", "wrong event data received")
+		return
+	}
+	var source blockSource
+	switch event.Events()[p2p.EventTypeKey][0] {
```

Comment:
`event.Events()[p2p.EventTypeKey]` are we guranteed this won't be empty?

Reply:
it shouldnt be empty but i added a check